include/gul14/traits.h

```diff
@@ -97,6 +90,212 @@ using remove_cvref_t = typename remove_cvref<T>::type;
 template <typename...>
 using void_t = void;
 
+
+//
+// invoke & friends
+//
+#if __cplusplus >= 201703L
+
+using std::invoke;
+using std::invoke_result;
+using std::invoke_result_t;
+using std::is_invocable;
+using std::is_invocable_r;
+
+#else
+
+namespace detail_invoke {
+
+#define GUL14_RETURN(...) \
```

Comment:
Now that it moved into a more generic header maybe the macro name is too broad and nondescript?

Otoh it lives only until line 187. But still.

Reply:
That is true, I have renamed it to GUL14_INVOKE_RETURN.